target/snitch_cluster/sw/apps/snax-streamer-gemm-conv/data/datagen.py

```diff
@@ -0,0 +1,512 @@
+    for mm in range(m):
+        for nn in range(n):
+            for kk in range(k):
+                for rr in range(row):
+                    for cc in range(col):
+                        for ss in range(size):
+                            c_index = (
+                                mm * n * row * col + nn * row * col + rr * col + cc
+                            )
+                            a_index = (
+                                mm * k * row * size + kk * row * size + rr * size + ss
+                            )
+                            b_index = (
+                                nn * k * size * col + kk * size * col + cc * size + ss
+                            )
+                            c[c_index] = c[c_index] + (a[a_index] - subtraction_a) * (
+                                b[b_index] - subtraction_b
+                            )
+    return c
+
+
+def data_reshuffler_golden_model(
+    tempLoop0,
+    tempLoop1,
+    spatial_len_0,
+    spatial_len_1,
+    tempStride0,
+    tempStride1,
+    spatialStride0,
+    spatialStride1,
+    data,
+    int32=False,
+):
+    # abstract illusion: k innermost loop, m second innermost loop,
+    # K third innermost loop, M outermost loop
+
+    # total loop bounds = spatial loop bounds * temporal loop bounds
+    K = tempLoop0 * spatial_len_0
+    M = tempLoop1 * spatial_len_1
+
+    # loop bounds settings
+    matrix_size = {"K": K, "M": M, "k": spatial_len_0, "m": spatial_len_1}
+
+    # stride settings
+    strides = {
+        "M": tempStride1,
+        "K": tempStride0,
+        "m": spatialStride1,
+        "k": spatialStride0,
+    }
+
+    if int32:
+        result_array = np.zeros((matrix_size["M"] * matrix_size["K"]), np.int32)
+    else:
+        result_array = np.zeros((matrix_size["M"] * matrix_size["K"]), np.int8)
+
+    # apply strided layout mapping for the golden model of data reshuffler
+    for M in range(matrix_size["M"] // matrix_size["m"]):
+        for K in range(matrix_size["K"] // matrix_size["k"]):
+            for m in range(matrix_size["m"]):
+                for k in range(matrix_size["k"]):
+                    result_array[
+                        # output address calculation with coutinued increment
+                        matrix_size["K"]
+                        // matrix_size["k"]
+                        * matrix_size["k"]
+                        * matrix_size["m"]
+                        * M
+                        + matrix_size["k"] * matrix_size["m"] * K
+                        + m * matrix_size["k"]
+                        + k
+                    ] = data[
+                        # input address calculation with
+                        # strided layout mapping eqaution
+                        strides["M"] * M
+                        + strides["K"] * K
+                        + strides["m"] * m
+                        + strides["k"] * k
+                    ]
+
+    return result_array.ravel()
```

Comment:
Is this not tested elsewhere, and can we thus not get that function from somewhere else?